qutebrowser/completion/models/urlmodel.py

```diff
@@ -81,6 +92,24 @@ def _add_quickmark_entry(self, name, url):
         """
         self.new_item(self._quickmark_cat, url, name)
 
+    def _add_bookmark_entry(self, title, url):
+        """Add a new bookmark entry to the completion.
+
+        Args:
+            title: The title of the new bookmark.
+            url: The URL of the new bookmark.
+        """
+        self.new_item(self._bookmark_cat, url, title)
+
+    def custom_filter(self, pattern, row, parent):
+        """Filter by url and title."""
+        index0 = self.index(row, 0, parent)
+        index1 = self.index(row, 1, parent)
+        url = self.data(index0) or ''
+        title = self.data(index1) or ''
+        return (pattern.casefold() in url.casefold() or pattern.casefold() in
+                title.casefold())
```

Comment:
Hmm... having custom filters is nice (thanks for adding that!) but in this case, wouldn't it make sense to rename `self.columns_to_highlight` to `columns_to_filter` or so, and implementing this in the generic filter in a generic way? It'd probably never make sense to highlight something but not filter it, right?